internal/controllers/core/cluster/reconciler.go

```diff
@@ -58,43 +55,51 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	err := r.ctrlClient.Get(ctx, nn, &obj)
 	if err != nil && !apierrors.IsNotFound(err) {
 		r.store.Dispatch(clusters.NewClusterDeleteAction(request.Name))
-		delete(r.connections, nn)
+		r.connManager.delete(nn)
 		return ctrl.Result{}, err
 	}
 
 	// The apiserver is the source of truth, and will ensure the engine state is up to date.
 	r.store.Dispatch(clusters.NewClusterUpsertAction(&obj))
 
-	connection, hasConnection := r.connections[nn]
+	var connectionChanged bool
```

Comment:
why bother tracking connectionChanged? is it just an optimization, or is it preventing some sort of loop?

Reply:
Removed in 08d3b3247f23443b985045bdd32fcfe4565095fb to unconditionally overwrite to make semantics simpler.

(It was mostly a vestige that didn't make as much sense as implementation grew - `Cluster` shouldn't be getting reconciled excessively so I'm not super concerned about thrashing the map)